src/string_scanner.cr

```diff
@@ -293,4 +386,38 @@ class StringScanner
     start = Math.min(Math.max(offset - 2, 0), Math.max(0, @str.size - 5))
     io << " \"" << @str[start, 5] << "\" >"
   end
+
+  # :nodoc:
+  class StringMatchData
```

Comment:
This seems to be a simple wrapper type with not mutable state.
So I figure it might as well be a struct to avoid an allocation?

```suggestion
  struct StringMatchData
```